modules/effects/arrow-effects/src/main/kotlin/arrow/effects/internal/ParallelUtils.kt

```diff
@@ -14,7 +14,7 @@ import kotlin.coroutines.experimental.suspendCoroutine
 /* See parMap3 */
 internal fun <F, A, B, C> Effect<F>.parMap2(ctx: CoroutineContext, ioA: Kind<F, A>, ioB: Kind<F, B>, f: (A, B) -> C,
   /* start is used because this has to start inside the coroutine. Using Future won't work */
-                                            start: (Kind<F, Unit>) -> Unit): Proc<C> = { cc ->
+                                            start: (Kind<F, Unit>) -> Unit): Proc<C> = { _, cc ->
```

Comment:
Cancellation needs to be setup here.

Reply:
`IOConnection` is not exposed through Async anymore, mimicking cats. Further explanations in a comment in the end.